controllers/machinehealthcheck_controller.go

```diff
@@ -151,7 +151,12 @@ func (r *MachineHealthCheckReconciler) Reconcile(req ctrl.Request) (_ ctrl.Resul
 
 	defer func() {
 		// Always attempt to patch the object and status after each reconciliation.
-		if err := patchHelper.Patch(ctx, m); err != nil {
+		// Patch ObservedGeneration only if the reconciliation completed successfully
+		patchOpts := []patch.Option{}
+		if reterr == nil {
+			patchOpts = append(patchOpts, patch.WithStatusObservedGeneration{})
```

Comment:
@fabriziopandini I'm actually wondering if it makes sense to have observedGeneration for MachineHealthCheck, since it's not really managing a resource directly, but rather interacting with externally managed resources (Machines). I think we should likely remove the field for MachineHealthCheck all together, thoughts?

Reply:
I don't have strong opinions, but `observedGeneration` for `MachineHealthCheck` could provide a signal about which spec/MHC settings the controller is acting upon, and thus help to determine if MHC is stuck or not.